release-content/0.15/release-notes/function_reflection.md

```diff
@@ -0,0 +1,224 @@
+In Bevy 0.15, functions can be reflected as well!
+
+This feature is opt-in and requires the `reflect_functions` feature to be enabled on `bevy`
+(or the `functions` feature on `bevy_reflect` if using that crate directly).
+
+It works by converting regular functions which arguments and return type derive [`Reflect`]
+into a [`DynamicFunction`] type using a new [`IntoFunction`] trait.
+
+```rust
+fn add(a: i32, b: i32) -> i32 {
+    a + b
+}
+
+let function = add.into_function();
+```
+
+With a `DynamicFunction`, we can then generate our list of arguments into an [`ArgList`]
+and call the function:
+
+```rust
+let args = ArgList::new()
+    .push_owned(25_i32)
+    .push_owned(75_i32);
+
+let result = function.call(args);
+```
+
+Calling a function returns a [`FunctionResult`] which contains our [`Return`] data 
+or a [`FunctionError`] if something went wrong.
+
+```rust
+match result {
+    Ok(Return::Owned(value)) => {
+        let value = value.try_take::<i32>().unwrap();
+        println!("Got: {}", value);
+    }
+    Err(err) => println!("Error: {:?}", err),
+    _ => unreachable!("our function always returns an owned value"),
+}
+```
+
+#### Closure Reflection
+
+This feature doesn't just work for regular functions—it works on closures too!
+
+For closures that capture their environment immutably, we can continue using `DynamicFunction`
+and `IntoFunction`. For closures that capture their environment mutably, there's
+[`DynamicFunctionMut`] and [`IntoFunctionMut`].
+
+```rust
+let mut total = 0;
+
+let increment = || total += 1;
+
+let mut function = increment.into_function_mut();
+
+function.call(ArgList::new()).unwrap();
+function.call(ArgList::new()).unwrap();
+function.call(ArgList::new()).unwrap();
+
+// Drop the function to release the mutable borrow of `total`.
+// Alternatively, our last call could have used `call_once` instead.
+drop(function);
+
+assert_eq!(total, 3);
+```
+
+#### `FunctionInfo`
+
+Reflected functions hold onto their type metadata via [`FunctionInfo`] which is automatically
+generated by the [`TypedFunction`] trait. This allows them to return information about the
+function including its name, arguments, and return type.
+
+```rust
+let info = String::len.get_function_info();
+
+assert_eq!(info.name().unwrap(), "alloc::string::String::len");
+assert_eq!(info.arg_count(), 1);
+assert!(info.args()[0].is::<&String>());
+assert!(info.return_info().is::<usize>());
+```
+
+One thing to note is that closures, anonymous functions, and function pointers
+are not automatically given names. For these cases, names can be provided manually.
+
+The same is true for all arguments including `self` arguments: names are not automatically
+generated and must be supplied manually if desired.
+
+Using `FunctionInfo`, a `DynamicFunction` will print out its signature when debug-printed.
+
+```rust
+dbg!(String::len.into_function());
+// Outputs:
+// DynamicFunction(fn alloc::string::String::len(_: &alloc::string::String) -> usize)
+```
```

Comment:
Just added this because I think it's really cool we can still print out the full signature, but let me know if it's too much to include for the release post.